sys/net/application_layer/coap/gcoap.c

```diff
@@ -135,7 +135,7 @@ static void _listen(sock_udp_t *sock)
         return;
     }
 
-    if (coap_get_code(&pdu) == COAP_CODE_EMPTY) {
+    if (pdu.hdr->code == COAP_CODE_EMPTY) {
```

Comment:
we might do that as well. i just thought it is a waste of the computation done in `coap_get_code` just to compare it against zero...

We could add `coap_get_code_raw()` to nanocoap...